staging/src/k8s.io/apimachinery/pkg/runtime/serializer/versioning/versioning_unstructured_test.go

```diff
@@ -336,3 +336,26 @@ func TestDecodeUnstructured(t *testing.T) {
 		assert.Equal(t, testCase.expectedGVKOfSerializedData, actualSerializedGVK, "%v failed", testCase.name)
 	}
 }
+
+func TestDecodeUnstructuredUntypedIntoTyped(t *testing.T) {
+	v1GVK := schema.GroupVersionKind{
+		Group:   "crispy",
+		Version: "v1",
+		Kind:    "Noxu",
+	}
+
+	json := []byte(`{items:[]}`)
+	into := &unstructured.UnstructuredList{}
+	gv, k := v1GVK.ToAPIVersionAndKind()
+	into.SetAPIVersion(gv)
+	into.SetKind(k + "List")
+
+	decoder := &mockSerializer{actual: &v1GVK, obj: into}
```

Comment:
use `into.DeepCopy()` here, otherwise this mock decoder triggers the short-circuit equality behavior